core/src/auctions.ts

```diff
@@ -347,7 +347,11 @@ export const bidWithCallee = async function (
     const calleeAddress = getCalleeAddressByCollateralType(network, auction.collateralType, marketId);
     const marketData = auction.marketDataRecords?.[marketId];
     const preloadedPools = marketData && 'pools' in marketData ? marketData.pools : undefined;
-    const calleeData = await getCalleeData(network, auction.collateralType, marketId, profitAddress, preloadedPools);
+    const oneInchData = marketData && 'oneInch' in marketData ? marketData.oneInch : undefined;
+    const calleeData = await getCalleeData(network, auction.collateralType, marketId, profitAddress, {
+        pools: preloadedPools,
+        oneInchParams: oneInchData ? { txData: oneInchData.calleeData, to: oneInchData.to } : undefined,
+    });
```

Comment:
The original proposal was:
- Remove `getCalleeData` from `bidWithCallee` function
- Execute `getCalleeData` inside `getMarketPrice` and store generated binary string (`calleeData`) inside `marketDataRecords[marketId]`, uniformly between callee types (ie, for all kind of callees)
  - The only additional value to replace `pools` that we will need to keep inside the `marketData` object will be `exchangePath` – I suggested to store generated string (like `LINK -> ETH -> DAI`)
  - This will remove the need for `pools` whatsoever
- Use pre-generated `calleeData` inside `bidWithCallee`

Please reach out to me/comment here if you didn't fully get the proposal

Reply:
> Execute getCalleeData inside getMarketPrice

then the `getMarketPrice` in this case does performs actions that do not correspond to its name which is e.g. problematic for `convertETHtoDAI` and/or all other functions (in `debt.ts`, `fees.ts`, etc) that only need the market price calculated.

> and store generated binary string (calleeData) inside marketDataRecords[marketId]

currently this function uses `getMarketDataRecords` call to receive this object (some of underlying functionality there is cached). So there is not object that is actually stored in the variable. So according to this proposal, i would have to instantiate such object, pass it around and manage its state which breaks the initial functional-procedural design paradigm.
  - E.g. i would have to return it from the `getMarketPrice` with all the `getCalleeData`s being called for every market (not only for the best one). 
  - After returning this object, i would have to propagate it upwards (or only the callee datas) to the enrichment functionality

If i interpret the consequences of the proposed approach correctly, then i do not see how the suggested here implementation is better than the one currently featured in the PR.

Reply:
the callee data generation requires the profit address. Should i fill this in and reencode this on the stage of bidding?

Reply:
True! But can you reencode it at bidding uniformly for all callees? (I think different callees have different order of arguments for that, right?) What would be another workaround?

Reply:
https://github.com/makerdao/exchange-callees/tree/master/src

all the callees there as of now comply with the position of profitAddress (i.e. first) so unpacking the data into address and bytes and packing it back should work up until we intend supporting a callee that violates this order.

___

i do not see the alternative workaround that is scoped to the core ( mb it's possible to also adjust store on frontend so it sends in the wallet address for querying the marketdata of the auctions, but this does not make sense because wallet should not be required for this)

Reply:
additional complications which seem to require more than just minor adjustments.

1. To implement the idea here, it would require to call the 1inch api as part of the `getCalleeData` for getting the `txData`. But this request requires the amount of exchanged token for the api. So we have the case where _generally_ unrelated information is to be passed to `getCalleeData`. And this edgecase is to be handled based on types. Just like now. So i dont think we're making something better in this scenario
2. Pools are provided in some cases by the market configuration, in other by the Alpharouter - along with the market price, there can be preloaded pools, there can be none. Removing the pools (if i got the idea correctly) from the scope of `getMarketPrice` also then would require additional logic put in every function to map this. And then changing the whole chain of `pools` transformations that propagates to frontend.

These both along with the mentioned `profitAddress` above seem to require expanding the functionality of the `getMarketPrice` and `getCalleeData` and these would have to account for a variety of cases/inputs. This is already the case, just in another form and less hacks that fill in a dummy value to replace it later.

I would suggest to leave the solution as it is now (given that i understood your intent correctly and am seeing some consequences that you did not account), then if there're code structure problems - open a refactoring issue, list the points that are problematic, discuss them, come up with the solutions that would work out and are less hacky. Currently i'm not seeing the benefit of the proposed approach, i also did not find the comment that highlights the imperfections of current implementation.

Reply:
Agreed in the daily: let's drop this requirement from this PR